socrates/lib/registration/index.js

```diff
@@ -46,28 +51,40 @@ app.get('/ical', function (req, res, next) {
 
 // TODO noch nicht freigeschaltete Funktionalitäten:
 
-app.post('/startRegistration', function (req, res, next) {
+function participate(registrationTupel, req, res, next) {
+  if (!req.user) { return res.redirect('/registration'); }
+  var member = req.user.member || new Member().initFromSessionUser(req.user, true);
+  subscriberstore.getSubscriber(member.id(), function (err, subscriber) {
+    if (err) { return next(err); }
+    res.render('participate', {member: member, addon: subscriber.addon()});
+  });
+}
 
-  // TODO was wenn derjenige nicht angemeldet ist? Soll trotzdem funktionieren!
-  var resourceName = 'single'; //req.params.resource;
-  var days = 'three';
-  registrationService.startRegistration(req.user.member.id(), currentUrl, resourceName, days, moment(), function (err, statusTitle, statusText) {
+app.post('/startRegistration', function (req, res, next) {
+  if (!isRegistrationOpen() || !req.body.nightsOptions) { return res.redirect('/registration'); }
+  var option = req.body.nightsOptions.split(',');
+  var registrationTupel = {activityUrl: req.body.activityUrl, resourceName: option[0], days: option[1]};
+  var memberId = req.user ? req.user.member.id() : 'SessionID' + req.sessionID;
+  registrationService.startRegistration(memberId, registrationTupel, function (err, statusTitle, statusText) {
     if (err) { return next(err); }
     if (statusTitle && statusText) {
       statusmessage.errorMessage(statusTitle, statusText).putIntoSession(req);
-      res.redirect('/registration');
-    } else {
-      res.redirect('/registration/completeRegistration');
+      return res.redirect('/registration');
+    }
+    if (!req.user) {
+      var returnToUrl = '/registration/participate';
+      req.session.registrationTupel = registrationTupel;
+      req.session.returnToUrl = returnToUrl;
+      return res.render('loginForRegistration', {returnToUrl: returnToUrl});
     }
+    participate(registrationTupel, req, res, next);
```

Comment:
Nachteil: Die URL wird nicht geändert
Nachteil: Ich finde das, was in dem POST passiert, unglaublich lang und schwer zu durchschauen

Aber wahrscheinlich hast Du Recht